recipe/build-cpp.sh

```diff
@@ -2,7 +2,11 @@
 
 set -ex
 
-if [[ "${abseil_cpp}" == "20210324.2" && "${target_platform}" == osx-* ]]; then
+echo CONFIG
+
+# Unconditionally set target platform for older abseil version,
+# when updating abseil relax CXX_STANDARD pin
+if [[ "${target_platform}" == osx-* ]]; then
   export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CXX_STANDARD=14"
 else
   export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CXX_STANDARD=17"
```

Comment:
That comment makes little sense to me (why "target platform", when the variable being set is `CMAKE_CXX_STANDARD`?). It's also dodgy to use a different C++ standard compared to what was used to compile abseil. Older abseil builds didn't encode that information, though for >=20211102 you can use the right version for the abseil you're using. However, this currently leads into other issues due to the way grpc declares its dependence on abseil (see #213), so I'm fine to let this pass if it works for you.